### PR TITLE
Update Nushell support to use supported $env update syntax

### PR DESCRIPTION
**Pull request by jntrnr**
Greetings!

The Nushell team is working on removing `let-env FOO = ...` and moving code to use `$env.FOO = ...` as it's more honest about what's happening (rather than being scoped like a `let`). This PR should move atuin to use the supported syntax (assuming I didn't miss anything 😅).

### CODE REVIEW

**Review by ellie — APPROVED** [review]
Hey! Thank you so much for letting us know proactively + also pushing out the fix 💖 

This totally makes sense - and should be everything that needs changing!

**Comment by happysalada** [comment]
@ellie would it be possible to make a release with this fix ?
For package maintainers, we either have to apply this patch manually or wait for a new release. The latter is easier for us, so I thought I would ask. No worries if you can't.

**Pull request merged**
